src/Text/Parsing/StringParser/String.purs

```diff
@@ -111,3 +116,38 @@ anyLetter = lowerCaseChar <|> upperCaseChar <?> "Expected a letter"
 -- | Match a letter or a number.
 alphaNum :: Parser Char
 alphaNum = anyLetter <|> anyDigit <?> "Expected a letter or a number"
+
+-- | Build the regular expression from the pattern and match it, ensuring
+-- | that the pattern only attempts to match from the start of the target.
+regex' :: String -> Parser String
+regex' pat =
+    case er of
+      Left _ ->
+        fail $ "Illegal regex " <> pat
+      Right r ->
+        regex r
+    where
+      pattern =
+        if startsWith "^" pat then
+          pat
+        else
+          "^" <> pat
+      er = Regex.regex pattern noFlags
```

Comment:
Can we inline this into the `case` expression please? I think it's a little easier to read that way.